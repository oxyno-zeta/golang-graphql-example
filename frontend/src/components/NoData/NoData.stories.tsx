import React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import NoData, { Props } from './NoData';

export default {
  title: 'Components/NoData',
  component: NoData,
} as Meta<typeof NoData>;

export const Playground: StoryFn<typeof NoData> = function C(args: Props) {
  return <NoData {...args} />;
};
