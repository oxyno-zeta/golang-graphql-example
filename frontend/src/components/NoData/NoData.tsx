import React from 'react';
import { useTranslation } from 'react-i18next';
import Typography, { TypographyProps } from '@mui/material/Typography';

export interface Props {
  typographyProps?: Omit<TypographyProps, 'children'>;
}

function NoData({ typographyProps = {} }: Props) {
  // Setup translate
  const { t } = useTranslation();

  return (
    <Typography
      sx={{ display: 'flex', justifyContent: 'center', textAlign: 'center', margin: '10px 0' }}
      {...typographyProps}
    >
      {t('common.noData')}
    </Typography>
  );
}

export default NoData;
