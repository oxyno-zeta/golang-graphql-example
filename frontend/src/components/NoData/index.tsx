import NoData, { Props as NoDataTypographyProps } from './NoData';

export default NoData;
export type { NoDataTypographyProps };
