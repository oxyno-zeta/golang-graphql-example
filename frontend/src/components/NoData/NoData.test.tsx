import React from 'react';
import { render } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import NoData from './NoData';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('NoData', () => {
  it('should display no data', async () => {
    const { container } = render(<NoData />);
    // Now find errors
    expect(container).toHaveTextContent('common.noData');
    expect(container).toMatchSnapshot();
  });

  it('should display no data with specific variant', async () => {
    const { container } = render(<NoData typographyProps={{ variant: 'body2' }} />);
    // Now find errors
    expect(container).toHaveTextContent('common.noData');
    expect(container.firstChild).toHaveClass('MuiTypography-body2');
    expect(container).toMatchSnapshot();
  });
});
