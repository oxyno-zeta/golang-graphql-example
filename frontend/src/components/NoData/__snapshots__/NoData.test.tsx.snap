// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NoData should display no data 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  margin: 10px 0;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body1 emotion-0"
  >
    common.noData
  </p>
</div>
`;

exports[`NoData should display no data with specific variant 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  margin: 10px 0;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body2 emotion-0"
  >
    common.noData
  </p>
</div>
`;
